src/lib.rs

```diff
@@ -160,4 +160,43 @@ mod tests {
         assert_eq!(bucket.consume(), true);
         assert_eq!(bucket.consume(), false);
     }
+
+    #[test]
+    fn consume_weight_over_time() {
+        let now = Rc::new(RefCell::new(Instant::now()));
+        let now_moved = now.clone();
+
+        let mut bucket = TokenBucket::new(3, Duration::from_secs(1))
+            .with_timer(Box::new(move || *now_moved.borrow()));
+
+        // consume all tokens at once
+        assert_eq!(bucket.consume_weight(3), true);
+        assert_eq!(bucket.consume_weight(1), false);
+
+        // sequentially consume tokens
+        *now.borrow_mut() += Duration::from_secs(1);
+        assert_eq!(bucket.consume_weight(2), true);
```

Comment:
@malor I'd like to ask for your opinion here. Please see my questions below.

1. Do you think `consume_weight()` is a good name, or shall we rename it into some `consume_with_weight()`?
2. Do you think we even need `consume()` and `consume_weight` functions? Or shall we proceed with just `consume()` that always receives a weight?

Reply:
I like the name `consume()`, but  I think it's a bit annoying that we have to support (and test!) two different versions of the function. So, maybe, let's proceed with `consume()` that takes the number of tokens? I don't think it's a big deal to pass 1.

Reply:
Agree. I'll make a patch.